### ISSUE TITLE
multi-gpu training issue

### ISSUE THREAD

**Issue by MiZhangWhuer** [pihalf/GeoFormer#4]
PyTorch Lighting is emplyed in the repo. However, it gets stucked when training with multple gpus. It seems that this issue occurs because the collate_fn in the training dataloader. How to fix it?

**Comment by pihalf** [comment]
Hi @MiZhangWhuer, allow me to look into this in the coming week, I'll get back to you

**Comment by MiZhangWhuer** [comment]
> Hi @MiZhangWhuer, allow me to look into this in the coming week, I'll get back to you

Hi, how about the status of multi-gpu training？Have you solved the stuck problem？

**Comment by MiZhangWhuer** [comment]
> Hi @MiZhangWhuer, allow me to look into this in the coming week, I'll get back to you

@pihalf Hi, how about the status of multi-gpu training？Is the the stuck problem been solved？

**Comment by pihalf** [comment]
Hi @MiZhangWhuer, I haven't had time to work on it yet, but I suspect the issue lies with the dynamic padding. I'll see if I can out a fix this weekend.

**Comment by MiZhangWhuer** [comment]
> Hi [@MiZhangWhuer](https://github.com/MiZhangWhuer), I haven't had time to work on it yet, but I suspect the issue lies with the dynamic padding. I'll see if I can out a fix this weekend.

@pihalf Hi, has the bug been fixed?

**Comment by pihalf** [comment]
Hi @MiZhangWhuer - Thank you again for your interest in our work, and for your insistence on keeping me at it! :) 

I've just pushed a new update to the repo which performs maximum length batching. It inevitably increases VRAM consumption, so you might want to control for samples of large sequence length, or modify this hyper parameter accordingly.

**Comment by MiZhangWhuer** [comment]
> Hi [@MiZhangWhuer](https://github.com/MiZhangWhuer) - Thank you again for your interest in our work, and for your insistence on keeping me at it! :)
> 
> I've just pushed a new update to the repo which performs maximum length batching. It inevitably increases VRAM consumption, so you might want to control for samples of large sequence length, or modify this hyper parameter accordingly.

@pihalf Thanks for the repaid reply, I will feedback to you as soon as the multi-gpu training is complished.

**Comment by MiZhangWhuer** [comment]
@pihalf Thanks again for your innovative work. The multi-gpu training works fine. I think that the pix2seq next token prediction schema will inspire the followings.

**Comment by pihalf** [comment]
Great, happy to hear. Closing this.
